### PR TITLE
Make 3scale batcher policy compatible with OIDC

### PR DESCRIPTION
**Pull request by kevprice83**
adds ability to report app_id only. Fixes [THREESCALE-1515](https://issues.jboss.org/browse/THREESCALE-1515)

### CODE REVIEW

**Comment by kevprice83** [comment]
@3scale/apicast-core still need to write tests for this...

**Comment by kevprice83** [comment]
@davidor I've pushed the integration test which at the moment is broken. Will need some help fixing that. Unit tests are passing fine.

**Comment by kevprice83** [comment]
@mikz @davidor I'm a bit stuck now trying to get the integration test working. Any help would be appreciated.

**Comment by mayorova** [comment]
@kevprice83 I've changed the tests, they now pass (I didn't check if the test would fail before your code change though).
Travis fails because of the docs - there are broken links. I've PR'ed the fix for master: https://github.com/3scale/apicast/pull/958

Once it's merged, I think it would be good to rebase it on top of master, and also use `master` as base, because I am not sure it will be included in 3.4, so `3.4-stable` may be confusing.

File: t/apicast-policy-3scale-batcher.t [hunk 1]
```diff
@@ -477,3 +483,135 @@ rewrite_by_lua_block {
...
+    content_by_lua_block {
+      ngx.req.read_body()
+      local post_args = ngx.req.get_post_args()
+      local app_id_match, usage_match
+      for k, v in pairs(post_args) do
+        if k == 'transactions[0][usage][hits]' then
+          usage_match = v == '2'
+        elseif k == 'transactions[0][app_id]' then
+          app_id_match = v == 'appid'
+        end
```
Comment (davidor) [inline t/apicast-policy-3scale-batcher.t]: Should we also make sure that `k` is never `transactions[0][app_key]` ? Just to be sure that we are not accidentally sending it also.

**Pull request merged**